driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -719,9 +719,7 @@ public void run() {
         // execute
         // the prepared query. So don't wait if no connection is available, simply abort.
         ListenableFuture<Connection> connectionFuture =
-            entry
-                .getValue()
-                .borrowConnection(0, TimeUnit.MILLISECONDS, 0, statement.getRoutingKey());
+            entry.getValue().borrowConnection(0, TimeUnit.MILLISECONDS, 0, null);
```

Comment:
Why not pass statement here but null instead?

Reply:
Because this is a prepare() method, so the query is never in conditions to be directed to a proper shard. At least this is what I noticed in my tests.